### PR TITLE
 nixos/no-x-libs: fix gst_all_1.gst_plugins_good 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

Tested by building ``libextractor`` with nox.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by figsoda** [comment]
I don't really know how to test this, unfortunately

**Comment by SuperSandro2000** [comment]
This can be tested by creating a nixosConfiguration which has ``environment.noXLibs = true`` and then by building ``nixosConfiguration.$NAME.pkgs.gst_all_1.gst_plugins_good``

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 1]
```diff
@@ -158,14 +158,18 @@ stdenv.mkDerivation rec {
...
-    "-Dximagesrc=disabled" # Linux-only
   ] ++ lib.optionals (!raspiCameraSupport) [
     "-Drpicamsrc=disabled"
   ];
 
   postPatch = ''
     patchShebangs \
       scripts/extract-release-date-from-doap-file.py
+  ''
+  # gstreamer-gl-* detection is broken due to an upstream meson issue
```
Comment (lilyinstarlight) [inline pkgs/development/libraries/gstreamer/good/default.nix:169]: Can you link the upstream meson issue in the comment here?

**Review by lilyinstarlight — COMMENTED** [review]
Looks good in principle, I added a few review comments and will test with this PR later to ensure no problems

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 2]
```diff
@@ -149,6 +147,8 @@ stdenv.mkDerivation rec {
     "-Dqt6=disabled"
   ] ++ lib.optionals (!gtkSupport) [
     "-Dgtk3=disabled"
+  ] ++ lib.optionals (!enableX11) [
+    "-Dximagesrc=disabled" # Linux-only
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/libraries/gstreamer/good/default.nix:151]: X11 is actually not linux only and some things can be compiled with x11 on darwin

File: pkgs/development/libraries/gstreamer/good/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/gstreamer/good/default.nix:169]: I don't know if there is some. There is a code comment https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/blob/discontinued-for-monorepo/meson.build#L321-324 which was added in https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/680

Comment (lilyinstarlight) [reply pkgs/development/libraries/gstreamer/good/default.nix:167]: Is this substituteInPlace even necessary? It seems to have been a while ago, but I just tried this PR without that substituteInPlace and it seems to have built fine (just saying gstreamer-gl was not found and moving on)

Comment (lilyinstarlight) [reply pkgs/development/libraries/gstreamer/good/default.nix:167]: (other than that question, this PR looks good now btw, thank you!)

Comment (SuperSandro2000) [reply pkgs/development/libraries/gstreamer/good/default.nix]: Just tried again and it seems to work without this now, too.

**Review by lilyinstarlight — APPROVED** [review]
Change looks good to me and I tested building with and without X libs on x86_64-linux and cross to aarch64-linux, and I also did a quick build with aarch64-darwin

Thanks!

**Pull request merged**